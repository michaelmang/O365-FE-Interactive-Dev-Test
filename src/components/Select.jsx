import classnames from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';

import styles from './Select.module.scss';
import currencies from '../currencies.js';

const Select = ({ label, onSelect, value }) => {
  let currencyOptions = currencies.map(c => (
    <option key={c} value={c}>{c}</option>
  ));

  return (
    <div className={classnames('field', styles.field)}>
      <div className="control">
        <div className="select">
          <select onChange={onSelect} value={value}>
            <option value="-1">{label}</option>
            {currencyOptions}
          </select>
        </div>
      </div>
    </div>
  );
};

Select.propTypes = {
  label: PropTypes.string.isRequired,
  onSelect: PropTypes.func.isRequired,
  value: PropTypes.string.isRequired,
};

export default Select;
