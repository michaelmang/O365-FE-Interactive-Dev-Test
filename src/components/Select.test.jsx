import { shallow } from 'enzyme';
import React from 'react';

import Select from './Select.jsx';
import renderSnapshot from '../testing/renderSnapshot.js';

jest.mock('./Select.module.scss', () => ({
  field: 'styles.field',
}));

describe('Select', () => {
  it('matches the saved snapshot', () => {
    let { root: actual } = extractComponents(renderComponent());

    expect(renderSnapshot(actual)).toMatchSnapshot();
  });

  describe('select', () => {
    it('passes down onChange prop with value of onSelect', () => {
      let onSelect = jest.fn();
      let { select: subject } = extractComponents(renderComponent({
        onSelect,
      }));

      expect(subject).toHaveProp(
        'onChange',
        onSelect,
      );
    });
  });

  const extractComponents = (root) => ({
    root,
    select: root.find('select'),
    someCurrencyOption: root.find('[value="BGN"]'),
  });

  const renderComponent = (props) => (
    shallow(
      <Select
        label="default label"
        onSelect={() => {}}
        value="default value"
        {...props}
      />
    )
  );
});
