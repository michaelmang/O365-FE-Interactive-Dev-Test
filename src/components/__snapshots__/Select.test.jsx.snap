// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select matches the saved snapshot 1`] = `
<div
  className="field styles.field"
>
  <div
    className="control"
  >
    <div
      className="select"
    >
      <select
        onChange={[Function]}
        value="default value"
      >
        <option
          value="-1"
        >
          default label
        </option>
        <option
          value="BGN"
        >
          BGN
        </option>
        <option
          value="NZD"
        >
          NZD
        </option>
        <option
          value="ILS"
        >
          ILS
        </option>
        <option
          value="RUB"
        >
          RUB
        </option>
        <option
          value="CAD"
        >
          CAD
        </option>
        <option
          value="USD"
        >
          USD
        </option>
        <option
          value="PHP"
        >
          PHP
        </option>
        <option
          value="CHF"
        >
          CHF
        </option>
        <option
          value="ZAR"
        >
          ZAR
        </option>
        <option
          value="AUD"
        >
          AUD
        </option>
        <option
          value="JPY"
        >
          JPY
        </option>
        <option
          value="TRY"
        >
          TRY
        </option>
        <option
          value="HKD"
        >
          HKD
        </option>
        <option
          value="MYR"
        >
          MYR
        </option>
        <option
          value="THB"
        >
          THB
        </option>
        <option
          value="HRK"
        >
          HRK
        </option>
        <option
          value="NOK"
        >
          NOK
        </option>
        <option
          value="IDR"
        >
          IDR
        </option>
        <option
          value="DKK"
        >
          DKK
        </option>
        <option
          value="CZK"
        >
          CZK
        </option>
        <option
          value="HUF"
        >
          HUF
        </option>
        <option
          value="GBP"
        >
          GBP
        </option>
        <option
          value="MXN"
        >
          MXN
        </option>
        <option
          value="KRW"
        >
          KRW
        </option>
        <option
          value="ISK"
        >
          ISK
        </option>
        <option
          value="SGD"
        >
          SGD
        </option>
        <option
          value="BRL"
        >
          BRL
        </option>
        <option
          value="PLN"
        >
          PLN
        </option>
        <option
          value="INR"
        >
          INR
        </option>
        <option
          value="RON"
        >
          RON
        </option>
        <option
          value="CNY"
        >
          CNY
        </option>
        <option
          value="SEK"
        >
          SEK
        </option>
      </select>
    </div>
  </div>
</div>
`;
